src/librustc_mir/transform/inline.rs

```diff
@@ -123,6 +123,13 @@ impl Inliner<'tcx> {
                     continue;
                 };
 
+                // Copy unevaluated constants
+                caller_body.required_consts.extend(
+                    callee_body.required_consts.iter().copied().filter(|&constant| {
+                        matches!(constant.literal.val, ConstKind::Unevaluated(_, _, _))
```

Comment:
We never add anything non-`Unevaluated` in there, do we?

Reply:
Fixed.

Reply:
No, but the above `subst_and_normalize_erasing_regions` may be able to evaluate a constant now. So after that we don't need to add it.

Reply:
Right I wasn't remembering why I've added that check, good to also add a comment explaining that.